packages/runtime/container-runtime/src/test/batching.spec.ts

```diff
@@ -138,106 +146,249 @@ describe("Runtime batching", () => {
+			batch[1].clientId = "otherClientId";
+
+			assert.throws(
+				() => processBatch(batch, containerRuntime),
+				(e: any) => {
+					assert(e.errorType === FluidErrorTypes.dataCorruptionError);
+					assert(e.message === "Received messages from multiple clients in a batch");
+					return true;
+				},
+				"Batch with ops from multiple clients should fail",
+			);
+		});
+
+		it("fails processing a batch containing a non-runtime op along with runtime ops", async () => {
+			const messageCount = 5;
+			const batch = getMessages(messageCount);
+			// Change the type of the second message to a non-runtime op.
+			batch[1].type = MessageType.NoOp;
+
+			assert.throws(
+				() => processBatch(batch, containerRuntime),
+				(e: any) => {
+					assert(e.errorType === FluidErrorTypes.dataProcessingError);
+					assert(e.message === "Received out-of-order messages in batch");
+					return true;
+				},
+				"Batch with non-runtime op along with runtime ops should fail",
+			);
+		});
+
+		it("fails processing a batch containing an unknown runtime op along with known ops", async () => {
+			const messageCount = 5;
+			const batch = getMessages(messageCount);
+
+			// Change the type of the second message to an unknown runtime op.
+			const unknownMessage = batch[1];
+			const unknownMessageType = "unknown";
+			(unknownMessage.contents as any).type = unknownMessageType;
+
+			assert.throws(
+				() => processBatch(batch, containerRuntime),
+				(e: any) => {
+					assert(e.errorType === FluidErrorTypes.dataProcessingError);
+					assert(e.message === "Runtime message of unknown type");
+					return true;
+				},
+				"Batch with unknown runtime op along with known ops should fail",
+			);
+		});
 	});
 
-	it("fails processing a batch containing a non-runtime op along with runtime ops", async () => {
-		const messageCount = 5;
-		const batch = getBatch(messageCount);
-		// Change the type of the second message to a non-runtime op.
-		batch[1].type = MessageType.NoOp;
-
-		assert.throws(
-			() => processBatch(batch, containerRuntime),
-			(e: any) => {
-				assert(e.errorType === FluidErrorTypes.dataProcessingError);
-				assert(e.message === "Received a system message during batch processing");
-				return true;
+	describe("Delta scheduler for batches", () => {
+		// Function to process an inbound op. It adds delay to simulate time taken in processing an op.
+		function processOp() {
+			// Add delay such that each op takes greater than the DeltaScheduler's processing time to process.
+			// The times increases every time a batch is processed so simulate by increasing the delay.
+			clock.tick(DeltaScheduler.processingTime + deltaSchedulerTimeIncrement);
+			deltaSchedulerTimeIncrement += DeltaScheduler.processingTimeIncrement;
+		}
+
+		let containerRuntimeStub: sinon.SinonStub;
+		let batchBeginCount = 0;
+		let batchEndCount = 0;
+		let deltaSchedulerTimeIncrement = DeltaScheduler.processingTimeIncrement;
```

Comment:
I'd not name this "increment", it's the sum of N increments.  So maybe "buffer" or something